spec/ddtrace/contrib/action_cable/patcher_spec.rb

```diff
@@ -0,0 +1,142 @@
+require 'spec_helper'
+require 'ddtrace'
+
+require 'ddtrace/contrib/analytics_examples'
+require 'rails'
+require 'active_support/core_ext/hash/indifferent_access'
+
+begin
+  require 'action_cable'
+rescue LoadError
+  puts 'ActionCable not supported in Rails < 5.0'
+end
+
+RSpec.describe 'ActionCable patcher' do
+  before { skip('ActionCable not supported') unless Datadog::Contrib::ActionCable::Integration.compatible? }
+
+  let(:tracer) { get_test_tracer }
+  let(:configuration_options) { { tracer: tracer } }
+
+  before do
+    Datadog.configure do |c|
+      c.use :action_cable, configuration_options
+    end
+  end
+
+  around do |example|
+    # Reset before and after each example; don't allow global state to linger.
+    Datadog.registry[:action_cable].reset_configuration!
+    example.run
+    Datadog.registry[:action_cable].reset_configuration!
+  end
+
+  let(:all_spans) { tracer.writer.spans(:keep) }
+
+  let(:span) do
+    expect(all_spans).to have(1).item
+    all_spans.find { |s| s.service == 'action_cable' }
+  end
+
+  context 'with server' do
+    let(:channel) { 'chat_room' }
+    let(:message) { 'Hello Internet!' }
+
+    let(:server) do
+      ActionCable::Server::Base.new.tap do |s|
+        s.config.cable = { adapter: 'inline' }.with_indifferent_access
+        s.config.logger = Logger.new(nil)
+      end
+    end
+
+    context 'on broadcast' do
+      subject(:broadcast) { server.broadcast(channel, message) }
+
+      it 'traces broadcast event' do
+        broadcast
+
+        expect(span.service).to eq('action_cable')
+        expect(span.name).to eq('action_cable.broadcast')
+        expect(span.span_type).to eq('custom')
+        expect(span.resource).to eq('action_cable.broadcast')
+        expect(span.get_tag('action_cable.channel')).to eq(channel)
+        expect(span.get_tag('action_cable.broadcast.coder')).to eq('ActiveSupport::JSON')
+        expect(span.status).to_not eq(Datadog::Ext::Errors::STATUS)
```

Comment:
Should this use that error helper you added?

Reply:
✅